test/e2e/member_operator_test.go

```diff
@@ -0,0 +1,115 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+	"github.com/codeready-toolchain/member-operator/pkg/common"
+	userv1 "github.com/openshift/api/user/v1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestUserAccount(t *testing.T) {
+	userAccList := &toolchainv1alpha1.UserAccountList{}
+
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, userAccList)
+	require.NoError(t, err, "failed to add custom resource scheme to framework")
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	// get global framework variables
+	f := framework.Global
+	client := f.Client.Client
+
+	// wait for operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "member-operator", 1, retryInterval, timeout)
+	require.NoError(t, err, "failed while waiting for operator deployment")
+
+	t.Log("member-operator is ready and running state")
+
+	// create useraccount
+	userAcc := newUserAcc(t, f, ctx)
+	err = f.Client.Create(context.TODO(), userAcc, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err)
+	t.Logf("user account '%s' created", userAcc.Name)
+
+	t.Run("verify_useraccount_ok", func(t *testing.T) {
+		err := verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_user_ok", func(t *testing.T) {
```

Comment:
Let's add some noise:
- Create another UserAccount and check if it's intact when manipulating with the main UserAccount. 
- Also create explicitly in the tests some noise Users and Identities (so they are not created by the controller). So, when we delete the UserAccount these ones should stay. It will add extra checks to our owner-secondary resource relations.

Reply:
done [574ed01](https://github.com/codeready-toolchain/member-operator/pull/26/commits/574ed010a31357d055086c84cea5565d269ae009)